python/hail/utils/tutorial.py

```diff
@@ -0,0 +1,164 @@
+import hail as hl
+from .java import FatalError, Env, info
+from .misc import local_path_uri, new_local_temp_dir
+import os
+import zipfile
+from urllib.request import urlretrieve
+
+__all__ = [
+    'get_1kg',
+    'get_movie_lens'
+]
+
+resources = {
+    '1kg_annotations': 'https://storage.googleapis.com/hail-tutorial/1kg_annotations.txt',
+    '1kg_matrix_table': 'https://storage.googleapis.com/hail-tutorial/1kg.vcf.bgz',
+    'movie_lens_100k': 'http://files.grouplens.org/datasets/movielens/ml-100k.zip',
+}
+
+tmp_dir: str = None
+
+
+def init_temp_dir():
+    global tmp_dir
+    if tmp_dir is None:
+        tmp_dir = new_local_temp_dir()
+
+
+def _mkdir(jhc, path):
+    if not Env.jutils().dirExists(jhc, path):
+        r = Env.jutils().mkdir(jhc, path)
+        if not r:
+            raise IOError(f'could not mkdir f{path}')
+
+
+def get_1kg(output_dir, overwrite: bool = False):
+    """Download public 1000 genotype data and sample annotations.
+
+    Parameters
+    ----------
+    output_dir
+        Directory in which to write data.
+    overwrite
+        If ``True``, overwrite existing files/directories at those locations.
+    """
+    jhc = Env.hc()._jhc
+
+    _mkdir(jhc, output_dir)
+
+    matrix_table_path = os.path.join(output_dir, '1kg.mt')
+    annotations_path = os.path.join(output_dir, '1kg_annotations.txt')
+
+    if (overwrite
+            or not Env.jutils().dirExists(jhc, matrix_table_path)
+            or not Env.jutils().fileExists(jhc, annotations_path)):
+        init_temp_dir()
+        tmp_vcf = os.path.join(tmp_dir, '1kg.vcf.bgz')
+        source = resources['1kg_matrix_table']
+        info(f'downloading 1KG VCF ...\n'
+             f'  Source: {source}')
+        urlretrieve(resources['1kg_matrix_table'], tmp_vcf)
+        cluster_readable_vcf = Env.jutils().copyToTmp(jhc, local_path_uri(tmp_vcf))
+        info('importing VCF and writing to matrix table...')
+        hl.import_vcf(cluster_readable_vcf, min_partitions=16).write(matrix_table_path, overwrite=True)
+
+        tmp_annot = os.path.join(tmp_dir, '1kg_annotations.txt')
+        source = resources['1kg_annotations']
+        info(f'downloading 1KG annotations ...\n'
+             f'  Source: {source}')
+        urlretrieve(source, tmp_annot)
+        hl.hadoop_copy(local_path_uri(tmp_annot), annotations_path)
+        info('Done!')
+    else:
+        info('1KG files found!')
+
+
+def get_movie_lens(output_dir, overwrite: bool = False):
+    """Download public Movie Lens dataset.
+
+    Notes
+    -----
+    See the
+    `MovieLens website <https://grouplens.org/datasets/movielens/100k/>`__
+    for more information about this dataset.
+
+    Parameters
+    ----------
+    output_dir
+        Directory in which to write data.
+    overwrite
+        If ``True``, overwrite existing files/directories at those locations.
```

Comment:
"overwrite any existing files/directories found at the corresponding locations."